### PR TITLE
Sign off jobs

### PR DESCRIPTION
**Pull request by kdelee**
@Spredzy I know we said I should do a mock up....

but I did a little bit more.

Screenshot from 2019-11-26 13-25-58

### CODE REVIEW

File: towerdashboard/app.py [hunk 1]
```diff
@@ -25,9 +25,10 @@
 
 def create_app():
     app = flask.Flask(__name__)
-    app.config.from_object('towerdashboard.settings.settings')
     if os.environ.get('TOWERDASHBOARD_SETTINGS'):
         app.config.from_envvar('TOWERDASHBOARD_SETTINGS')
+    else:
```
Comment (elyezer) [inline towerdashboard/app.py]: Why is this needed? The previous way was to trying to load from default settings and override with any environment settings if any is available.

File: towerdashboard/templates/jenkins/releases.html [hunk 2]
```diff
@@ -171,7 +171,54 @@ <h4>Integration Results</h4>
             </tr>
         </tbody>
     </table>
-
+    <table class="table table-bordered">
```
Comment (elyezer) [inline towerdashboard/templates/jenkins/releases.html]: I think it would be good to have a title or a separator here. On the screenshot you shared we have the usual table plus this one in sequence but it is not clear at first or without some context that it is a different table. What do you think?

File: towerdashboard/app.py (hunk 1 above)
Comment (kdelee) [reply towerdashboard/app.py]: for some reason this was breaking when I tried to run locally the other way around. This way I figure it loads from my `TOWERDASHBOARD` var first if it exists, and then if not it does the other...so should not be behavior change on prod I think

File: towerdashboard/templates/jenkins/releases.html (hunk 2 above)
Comment (kdelee) [reply towerdashboard/templates/jenkins/releases.html]: yeah that sounds good...I can try and do something like there is w/ the Integration Results seperator
Screenshot from 2019-11-26 15-51-31

File: README.md [hunk 3]
```diff
@@ -157,6 +157,26 @@ To be able to set up a dev environment, few steps are required.
...
+These are unique on combination of the parameters:
+	1)  `"tower"` matching a `"code"` in the list of `TOWER_VERSIONS` found in the [base data used](https://github.com/ansible/tower-dashboard/blob/master/towerdashboard/data/base.py)
+	1)  `"component"` matching an item in the list of `SIGN_OFF_COMPONENTS` found in the [base data used](https://github.com/ansible/tower-dashboard/blob/master/towerdashboard/data/base.py)
+	1)  `"deploy"` matching a value of `"deploy"` in the list of `SIGN_OFF_DEPLOYMENTS` found in the [base data used](https://github.com/ansible/tower-dashboard/blob/master/towerdashboard/data/base.py), e.g. `"standalone"` or `"cluster"`.
+	1)  `"tls"` matching a value of `"tls"` in the list of `SIGN_OFF_DEPLOYMENTS` found in the [base data used](https://github.com/ansible/tower-dashboard/blob/master/towerdashboard/data/base.py), e.g. `True` or `False`.
+	1)  `"platform"` matching an item in the list of `SIGN_OFF_PLATFORMS` found in the [base data used](https://github.com/ansible/tower-dashboard/blob/master/towerdashboard/data/base.py), e.g. `rhel-7.7-x86_64` or `OpenShift`.
```
Comment (kdelee) [inline README.md]: looks like this is not quite rendering correctly --need to make the list actually look like a list. May need newlines between them

**Comment by kdelee** [comment]
added line break for new sign off jobs section
Screenshot from 2019-12-02 15-12-51

File: README.md [hunk 4]
```diff
@@ -157,6 +157,31 @@ To be able to set up a dev environment, few steps are required.
...
+1)  `"tower"` matching a `"code"` in the list of `TOWER_VERSIONS` found in the [base data used](https://github.com/ansible/tower-dashboard/blob/master/towerdashboard/data/base.py)
+
+1)  `"component"` matching an item in the list of `SIGN_OFF_COMPONENTS` found in the [base data used](https://github.com/ansible/tower-dashboard/blob/master/towerdashboard/data/base.py)
+
+1)  `"deploy"` matching a value of `"deploy"` in the list of `SIGN_OFF_DEPLOYMENTS` found in the [base data used](https://github.com/ansible/tower-dashboard/blob/master/towerdashboard/data/base.py), e.g. `"standalone"` or `"cluster"`.
+
+1)  `"tls"` matching a value of `"tls"` in the list of `SIGN_OFF_DEPLOYMENTS` found in the [base data used](https://github.com/ansible/tower-dashboard/blob/master/towerdashboard/data/base.py), e.g. `True` or `False`.
+
+1)  `"platform"` matching an item in the list of `SIGN_OFF_PLATFORMS` found in the [base data used](https://github.com/ansible/tower-dashboard/blob/master/towerdashboard/data/base.py), e.g. `rhel-7.7-x86_64` or `OpenShift`.
+
```
Comment (kdelee) [inline README.md]: BAH! I have realized I am missing ansible version as a component here....going to work on that

File: towerdashboard/data/base.py [hunk 5]
```diff
@@ -172,6 +172,28 @@
...
+    'rhel-7.7-x86_64',
+    'rhel-8.1-x86_64',
+    'OpenShift'
+]
+
+SIGN_OFF_DEPLOYMENTS = [
+    {'deploy': 'cluster', 'tls': True},
+    {'deploy': 'cluster', 'tls': False},
+    {'deploy': 'standalone', 'tls': True},
+    {'deploy': 'standalone', 'tls': False}
```
Comment (Spredzy) [inline towerdashboard/data/base.py]: Should FIPS true/false be present here too ?

Comment (elyezer) [reply towerdashboard/data/base.py]: If we are adding more permutation here I would suggest to let python build the list for us:

```python
>>> [
    {'deploy': deploy, 'tls': tls, 'fips': fips}
    for deploy, tls, fips in itertools.product(
        ('standalone', 'cluster'),
        (True, False),
        (True, False),
    )
]
        
[{'deploy': 'standalone', 'tls': True, 'fips': True},
 {'deploy': 'standalone', 'tls': True, 'fips': False},
 {'deploy': 'standalone', 'tls': False, 'fips': True},
 {'deploy': 'standalone', 'tls': False, 'fips': False},
 {'deploy': 'cluster', 'tls': True, 'fips': True},
 {'deploy': 'cluster', 'tls': True, 'fips': False},
 {'deploy': 'cluster', 'tls': False, 'fips': True},
 {'deploy': 'cluster', 'tls': False, 'fips': False}]

File: README.md [hunk 6]
```diff
@@ -157,6 +157,33 @@ To be able to set up a dev environment, few steps are required.
 tower-dashboard should be running on your local loop on port 5000 (`http://127.0.0.1:5000`)
 
 
+## Sending data to the dashboard
+
+### Top level results
+
+FIXME
```
Comment (elyezer) [inline README.md]: Are we fixing this on the PR or is this supposed to come later?

File: towerdashboard/data/base.py (hunk 5 above)
Comment (Spredzy) [reply towerdashboard/data/base.py]: Yep. Way more readable. My ultimate concern here is  how exhausstive do we want to be. Adding any new bool will trigger N cases of more scenario to testt and that might not be meaninful (while full exhaustive). I think we need to come with a "strategic" approach of "What is it we want to test?" (ie. what are the most common scenarios used our there). So we can via our testing cover 90%+ used scenario and ship with confidence.

Comment (elyezer) [reply towerdashboard/data/base.py]: Yeah, maybe coming with with a subset of the permutation it would give us the same reliability as of running all the permutations. In case we decide to run all of the permutations that snippet might be useful.

File: README.md (hunk 6 above)
Comment (kdelee) [reply README.md]: I can give it a stab, I just wanted to document how to POST data into the system and started with my part but saw obvious need to also document for the main matrix

File: README.md [hunk 7]
```diff
@@ -157,6 +157,33 @@ To be able to set up a dev environment, few steps are required.
...
+```bash
+curl -d '{"tower":"devel", "url":"https://your.job.runner.com/job/1", "component":"install", "status":"FAILURE", "tls":0, "deploy":"standalone", "platform":"rhel-7.7-x86_64", "ansible": "devel"}' -H 'Content-t
+ype: application/json' -X POST http://127.0.0.1:5000/jenkins/sign_off_jobs
+```
+This will correlate it to one of the jobs created at database initializaion time and update the latest result.
+These are unique on combination of the parameters:
+
+1)  `"tower"` matching a `"code"` in the list of `TOWER_VERSIONS` found in the [base data used](https://github.com/ansible/tower-dashboard/blob/master/towerdashboard/data/base.py)
+
+1)  `"component"` matching an item in the list of `SIGN_OFF_COMPONENTS` found in the [base data used](https://github.com/ansible/tower-dashboard/blob/master/towerdashboard/data/base.py)
```
Comment (elyezer) [inline README.md]: Should these ordered list numbers be changed?

File: README.md (hunk 6 above)
Comment (elyezer) [reply README.md]: This can be done later, maybe having an issue to remember that is for now

File: towerdashboard/app.py [hunk 8]
```diff
@@ -25,9 +25,10 @@
...
+        app.config.from_object('towerdashboard.settings.settings')
```
Comment (dsesami) [inline towerdashboard/app.py]: double check to make sure this isn't actually `towerdashboard.settings`. I see that exists in my env, but not `towerdashboard.settings.settings`.

File: README.md (hunk 7 above)
Comment (kdelee) [reply README.md]: they auto number when markdown is rendered AFAIk will check

Comment (kdelee) [reply README.md]: yup, looks good when rendered

**Comment by dsesami** [comment]
some touches of my own. 
> reformatted date/time to locale-friendly format
> Highlight issues with no owner
> add table headers.

Please feel free to change the color of the highlighting if you prefer.

Screenshot from 2019-12-16 16-25-19

File: towerdashboard/data/base.py (hunk 5 above)
Comment (kdelee) [reply towerdashboard/data/base.py]: added fips and we can pear down the list at a future date :dancer:

File: README.md (hunk 7 above)
Comment (kdelee) [reply README.md]: markdown renders fine, autonumbering the entries

**Comment by kdelee** [comment]
@dsesami looks good to me, easy PR if someone has new thoughts.

I have now completed the TODO items I discussed w/ @Spredzy 

```
 - use same data but display 3 tables DONE
    - unstable results DONE
    - failed jobs DONE
    - a collapsible table of all sign off jobs DONE
 - make tls just a string "true" or "false" DONE
 - add fips, string value "true" or "false" DONE (openshift always false)
```

**Comment by kdelee** [comment]
here's a screen shot of lower portion w/ some stable/unstable results and collapsed sign off jobs area
Screenshot from 2019-12-16 17-27-02

File: towerdashboard/app.py (hunk 8 above)
Comment (dsesami) [reply towerdashboard/app.py]: Confirmed with @Spredzy that this is just how the RPM is set up. the settings file is symlinked into /usr/lib/python2.7/site-packages/towerdashboard/settings/settings.py

**Comment by kdelee** [comment]
meeeerge
